### PR TITLE
Update Kubeaddons to v0.7.4

### PR DESCRIPTION
**Pull request by shaneutt**
This update fixes bugs related to Kudo and also includes changes to automate

### CODE REVIEW

**Comment by shaneutt** [comment]
I've pulled in the testing code from [kubernetes-base-addons](https://github.com/mesosphere/kubernetes-base-addons) and adjusted it for the latest release of Kubeaddons.

[DCOS-63260](https://jira.mesosphere.com/browse/DCOS-63260) is the follow up to port these common tools into the framework itself in later iterations.

File: test/addons_test.go [hunk 1]
```diff
@@ -1,103 +1,256 @@
...
+	patchStorageClass        = `{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"false"}}}`
+)
+
+var addonTestingGroups = make(map[string][]string)
+
+func init() {
+	b, err := ioutil.ReadFile("groups.yaml")
 	if err != nil {
-		t.Fatal(err)
+		panic(err)
```
Comment (joejulian) [inline test/addons_test.go]: Why are we using panic instead of t.Fatal?

Comment (shaneutt) [reply test/addons_test.go]: The diff is bad, this line was not actually changed for this.

Comment (joejulian) [reply test/addons_test.go]: Ah, one of _those_ PRs. :stuck_out_tongue_closed_eyes:

**Comment by shaneutt** [comment]
> > "fixes bugs" and "includes changes"
> 
> Can this be split into two PRs?

Not easily, the changes and the bug fixes come as a single package, getting one without the other would require significant git wizardy.

**Pull request merged**